### PR TITLE
Update `hasRequiredScopes` function

### PR DESCRIPTION
**Pull request by Achintha444**
### Purpose
$Subject

### Checklist
- [ ] e2e cypress tests locally verified.
- [ ] Manual test round performed and verified.
- [ ] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there are any)
- [ ] Unit tests provided. (Add links if there are any)
- [ ] Integration tests provided. (Add links if there are any)

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: apps/console/src/features/authorization/models/authorization.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/**
+ * Interface to store authorization details.
+ */
+export interface AuthorizationInterface {
```
Comment (brionmario) [inline apps/console/src/features/authorization/models/authorization.ts:22]: Lets move this to `[use-authorization.t](https://github.com/wso2/identity-apps/pull/4368/files#diff-12e93760b28bfb2e85c786ed5f9b351fd41477e763024337bcb3846a11e9ba27)` as `UseAuthorizationInterface`.

**Review by brionmario — APPROVED** [review]

File: apps/console/src/features/authorization/models/authorization.ts (hunk 1 above)
Comment (Achintha444) [reply apps/console/src/features/authorization/models/authorization.ts]: Fixed with https://github.com/wso2/identity-apps/pull/4368/commits/247aec5782d7ace24b15cfe9e857299100328b4a

File: apps/console/src/features/authorization/hooks/use-authorization.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import getLegacyAuthzRuntime from "../utils/get-legacy-authz-runtime";
+
+/**
+ * Interface to store use authorization hook details.
+ */
+export interface UseAuthorizationInterface {
```
Comment (dasuni-30) [inline apps/console/src/features/authorization/hooks/use-authorization.ts:24]: Shall we move this to models?

File: modules/core/src/helpers/access-control.ts [hunk 3]
```diff
@@ -79,23 +112,35 @@ export const hasRequiredScopes = (
...
  */
-export const isPortalAccessGranted = <T = {}>(featureConfig: T, allowedScopes: string): boolean => {
-    const isDefined = featureConfig && !isEmpty(featureConfig);
+export const isPortalAccessGranted = <T = unknown>(
+    featureConfig: T, 
+    allowedScopes: string, 
+    organzationType?: string, 
+    isLegacyRuntimeDisabled?: boolean
+): boolean => {
+    const isDefined: boolean = featureConfig && !isEmpty(featureConfig);
```
Comment (dasuni-30) [inline modules/core/src/helpers/access-control.ts:128]: Do we need a separate constant for this?

File: apps/console/src/features/authorization/hooks/use-authorization.ts (hunk 2 above)
Comment (Achintha444) [reply apps/console/src/features/authorization/hooks/use-authorization.ts:24]: Moved to this file based on the above comment, https://github.com/wso2/identity-apps/pull/4368#discussion_r1373661010

File: modules/core/src/helpers/access-control.ts (hunk 3 above)
Comment (Achintha444) [reply modules/core/src/helpers/access-control.ts:128]: Did this for readability

**Review by dasuni-30 — APPROVED** [review]

**Pull request merged**